Pcan numerical channel

XelaRellum opened the pull request:
Allow the pcan channel to be an int. Also limit the channel to valid channel names.

KloolK commented on can/interfaces/pcan/pcan.py:
```diff
@@ -46,36 +51,108 @@
         :param str channel:
-            The can interface name. An example would be 'PCAN_USBBUS1'
+            The can interface name. An example would be 'PCAN_USBBUS1'.
+            Alternatively the value can an int with the numerical value.
```
Small typo: [...] can **be** an int [...]

XelaRellum commented:
i wanted to provide a unit-test, but i was unsure about test setup scenario and the possible mocking solutions.

hardbyte commented:
Thanks for the contribution!

Anyone with a pcan device able to verify this PR?

XelaRellum commented:
> 
> 
> Thanks for the contribution!
> 
> Anyone with a pcan device able to verify this PR?
Probably not ideally if the person doing the change also verifying this. Anyway, for what it's worth, I attached a test script that opens the interface first by name and then by number and then receive a single message (I had a device attached).

[test_pcan.log](https://github.com/hardbyte/python-can/files/6107629/test_pcan.log)


[test_pcan.zip](https://github.com/hardbyte/python-can/files/6107636/test_pcan.zip)

KloolK commented:
I only have a single pcan interface at home right now, so I can't test the receiving portion. But setting up the interface seems to work fine:
```
INFO:test_bcan:Open interface by name
INFO:test_bcan:Wait for CAN message
INFO:test_bcan:None
INFO:test_bcan:Open interface by number
INFO:test_bcan:Wait for CAN message
INFO:test_bcan:None
```

I'll gladly do more testing once I get the second interface (probably next week or the week after).

KloolK commented:
I ran your test with two PCAN-USB FD interfaces:
```
INFO:test_bcan:Open interface by name
INFO:test_bcan:Wait for CAN message
INFO:test_bcan:Timestamp:    15281.221878        ID: 07e0    S                DLC:  8    02 3e 00 00 00 00 00 00
INFO:test_bcan:Open interface by number
INFO:test_bcan:Wait for CAN message
INFO:test_bcan:Timestamp:    15281.860340        ID: 07e0    S                DLC:  8    02 3e 00 00 00 00 00 00
```
As far as that test goes, it seems fine.

I noticed two other things:
1. It seems like this entire PR is based on python-can 3.1. Can you please rebase onto the current release?
2. When using a numerical channel, `self.channel_info` becomes an `int`, which leads to `self.__str__` returning an `int`. I guess you have to convert to a `str` and store that as `self.channel_info` instead.

hardbyte commented:
Thanks @KloolK - Please let me know when you're happy to merge.

XelaRellum commented:
> I ran your test with two PCAN-USB FD interfaces:
> 
> ```
> INFO:test_bcan:Open interface by name
> INFO:test_bcan:Wait for CAN message
> INFO:test_bcan:Timestamp:    15281.221878        ID: 07e0    S                DLC:  8    02 3e 00 00 00 00 00 00
> INFO:test_bcan:Open interface by number
> INFO:test_bcan:Wait for CAN message
> INFO:test_bcan:Timestamp:    15281.860340        ID: 07e0    S                DLC:  8    02 3e 00 00 00 00 00 00
> ```
> 
> As far as that test goes, it seems fine.
> 
> I noticed two other things:
> 
>     1. It seems like this entire PR is based on python-can 3.1. Can you please rebase onto the current release?
> 
>     2. When using a numerical channel, `self.channel_info` becomes an `int`, which leads to `self.__str__` returning an `int`. I guess you have to convert to a `str` and store that as `self.channel_info` instead.

I rebased my changes on _develop_ and added the proposed change for _channel_info_

KloolK commented:
Tested the recent changes and everything seems fine.
@hardbyte I'd say it's ready to be merged.
